### PR TITLE
Practice2

### PR DESCRIPTION
**Pull request by nick-krasnitskiy**
1. Integrate SwiftLint into the project. Add some disabled rules to swiftlint.yml.
2. Create base UI using Compositional Layout.
3. Customize cell configurations using xib-files.
4. Create a model and manager for URLSession and JSON parsing. 
5. Configure the cells. Transfer API data to images and labels.
6. Add DispatchQueue.
6. Add app icons.
7. Configure the Launch Screen.
8. Add .gitignore file.
9. Fix Swiftlint warnings.

### CODE REVIEW

File: DogsList/DogsList/Controllers/ListDogViewController.swift [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        }
+        
+        return layout
+    }
+    
+    private func configureDataSource() {
+        dataSource = DataSource(collectionView: collectionView, cellProvider: { (collectionView, indexPath, imageURL) -> UICollectionViewCell? in
+            
+            switch indexPath.section {
+            case 0:
```
Comment (alekseykorol) [inline DogsList/DogsList/Controllers/ListDogViewController.swift]: Please use enum for sections

File: DogsList/DogsList/Controllers/ListDogViewController.swift [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    func stopActivityIndicator() {
+        DispatchQueue.main.async {
+            self.indicator.hidesWhenStopped = true
+            self.indicator.stopAnimating()
+            self.collectionView.isHidden = false
+        }
+    }
+    
+    func didFailWithError(error: Error) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
```
Comment (alekseykorol) [inline DogsList/DogsList/Controllers/ListDogViewController.swift:171]: What is the delay for?

File: DogsList/DogsList/Model/DogManager.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
+//
+//  DogManager.swift
+//  DogsList
+//
+//  Created by Nick Krasnitskiy on 06.07.2021.
+//
+
+import Foundation
+
+protocol DogManagerDelegate: class {
```
Comment (alekseykorol) [inline DogsList/DogsList/Model/DogManager.swift]: Using 'class' keyword for protocol inheritance is deprecated; use 'AnyObject' instead

**Review by alekseykorol — CHANGES_REQUESTED** [review]
Looks good!
It's very good that you tried UICollectionViewDiffableDataSource & UICollectionViewCompositionalLayout. 
Please fix all comments and finalize practice 3.

File: DogsList/DogsList/Controllers/ListDogViewController.swift (hunk 1 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList/Controllers/ListDogViewController.swift]: Thanks. Ok.

File: DogsList/DogsList/Model/DogManager.swift (hunk 3 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList/Model/DogManager.swift]: Thanks. Ok.

File: DogsList/DogsList/Controllers/ListDogViewController.swift (hunk 2 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList/Controllers/ListDogViewController.swift:171]: Thanks. The delay allows the alert does not overlap the activity indicator.

**Review by alekseykorol — APPROVED** [review]

**Pull request merged**